docs/manipulation.rst

```diff
@@ -102,6 +102,64 @@ Note: This method is not flux conserving.
     >>> spec1 = Spectrum1D(spectral_axis=np.arange(1, 50) * u.nm, flux=np.random.sample(49) * u.Jy)
     >>> spec1_msmooth = median_smooth(spec1, width=3)
 
+Resampling
+----------
+:ref:`specutils <specutils>` contains several classes for resampling the flux
+in a :class:`~specutils.Spectrum1D` object.  Currently supported methods of
+resampling are integrated flux conserving with :class:`~specutils.manipulation.FluxConservingResample`,
+linear interpolation with :class:`~specutils.manipulation.LinearInterpolatedResample`,
+and cubic spline with :class:`~specutils.manipulation.SplineInterpolatedResample`.
+Each of these classes takes in a :class:`~specutils.Spectrum1D` and a user
+defined output dispersion grid, and returns a new :class:`~specutils.Spectrum1D`
+with the resampled flux. Currently the resampling classes expect the new
+dispersion grid unit to be the same as the input spectrum's dispersion grid unit.
+
+If the input :class:`~specutils.Spectrum1D` contains an uncertainty,
+:class:`~specutils.manipulation.FluxConservingResample` will propogate the
+uncertainty to the final output :class:`~specutils.Spectrum1D`. However, the
+other two implemented resampling classes (:class:`~specutils.manipulation.LinearInterpolatedResample`
+and :class:`~specutils.manipulation.SplineInterpolatedResample`) will ignore
+any input uncertainty.
+
+Here's a set of simple examples showing each of the three types of resampling:
+
+.. code-block:: python
+
+    >>> from specutils import Spectrum1D
+    >>> import astropy.units as u
+    >>> import numpy as np
+    >>> from specutils.manipulation import FluxConservingResample, LinearInterpolatedResample, SplineInterpolatedResample
+
+    >>> input_spec = Spectrum1D(spectral_axis=np.arange(1, 20, 2) * u.nm, flux=np.random.sample(10)*u.Jy)
```

Comment:
This is not an especially realistic spectrum.  Two possibilities I'd like to suggest:
1) Use a "real" spectrum, like the one that comes from the "gettting started" on the front of the docs, and resample that.  Then instead of displaying the results as numbers, you'd show *plots* of the spectrum (perhaps zoomed in) to demonstrate the resampling in action.
2) Have the wavelength range be more like 400 to 700 nm.  There aren't very many spectra out there that are 10s of nanometers (but plenty that are in the visible range)

1 is a bit more useful, but 2 is a much easier change.  So one option might be to do 2 for now, and then in a follow-on PR perhaps I or someone else can try to implement 1.